prov/gni/src/gnix_msg.c

```diff
@@ -407,6 +505,71 @@ static int __gnix_rndzv_req_complete(void *arg, gni_return_t tx_status)
 	return ret;
 }
 
+/*
+ * As the completer fn is called in the nic's progress loop, the remote
+ * side's fabric request will keep track of the state of this rndzv iov
+ * transaction, once the outstanding_txds reaches zero for this fabric
+ * request, we are ready to generate CQEs and send the fin msg back to the
+ * sender
+ */
+static int __gnix_rndzv_iov_req_complete(void *arg, gni_return_t tx_status)
+{
+	struct gnix_tx_descriptor *txd = (struct gnix_tx_descriptor *)arg;
+	struct gnix_fab_req *req = txd->req;
+	int i, ret = FI_SUCCESS;
+
+	GNIX_TRACE(FI_LOG_EP_DATA, "\n");
+
+	GNIX_DEBUG(FI_LOG_EP_DATA, "req->msg.outstanding_txds = %d\n",
+		   atomic_get(&req->msg.outstanding_txds));
+
+	if (tx_status != GNI_RC_SUCCESS) {
+		req->tx_failures++;
```

Comment:
This retransmission code doesn't seem right.  It looks like the request will be queued to be resent as soon as any one of the post descriptors fails.  If many post descriptors fail the request would be queued many times.